src/TransportClient/Curl/HttpClient.php

```diff
@@ -86,4 +90,50 @@ public function post(string $url, ?array $payload): TransportClientResponse
             return new TransportClientResponse($httpStatus, $response);
         }
     }
+
+    private function readVersionFile()
+    {
+        // Define the path to the .version file
+        $versionFilePath = __DIR__ . '/../../../.version';
+
+        // Check if the file exists
+        if (!file_exists($versionFilePath)) {
+            // Handle the error if the file does not exist
+            throw new TransportRequestException("The .version file does not exist.");
+        }
+
+        // Read the contents of the file
+        $version = file_get_contents($versionFilePath);
+
+        // Check if the file was read successfully
+        if ($version === false) {
+            // Handle the error if the file could not be read
+            throw new TransportRequestException("Could not read the .version file.");
+        }
+
+        // Trim any whitespace from the beginning and end of the version string
+        $version = trim($version);
+
+        // Validate the version string
+        if (!$this->isSemanticVersion($version)) {
+            // Handle the error if the file is not valid
+            throw new TransportRequestException("The version in the .version file is not a valid Semantic Versioning compliant version.");
+        }
+
+        return $version;
+    }
+
+    /**
+     * Checks if the given string is a version that conforms to Semantic Versioning.
+     */
+    private function isSemanticVersion(string $version): bool
+    {
+        $pattern = '/^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/';
```

Comment:
I would suggest to simplify this: 

`return (bool) preg_match('/^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/',
        $version);`